packages/google_sign_in/google_sign_in/example/ios/RunnerTests/GoogleSignInTests.m

```diff
@@ -26,62 +26,406 @@ - (void)setUp {
+      [FlutterMethodCall methodCallWithMethodName:@"init"
+                                        arguments:@{@"hostedDomain" : [NSNull null]}];
+
+  XCTestExpectation *expectation = [self expectationWithDescription:@"expect result returns true"];
+  [self.plugin handleMethodCall:methodCall
+                         result:^(id r) {
+                           [expectation fulfill];
+                         }];
+  [self waitForExpectationsWithTimeout:5.0 handler:nil];
+  OCMVerify([self.mockSharedInstance setHostedDomain:nil]);
+}
+
+- (void)testInitDynamicClientId {
+  FlutterMethodCall *methodCall =
+      [FlutterMethodCall methodCallWithMethodName:@"init"
+                                        arguments:@{@"clientId" : @"mockClientId"}];
+
+  XCTestExpectation *expectation = [self expectationWithDescription:@"expect result returns true"];
+  [self.plugin handleMethodCall:methodCall
+                         result:^(id r) {
+                           [expectation fulfill];
+                         }];
+  [self waitForExpectationsWithTimeout:5.0 handler:nil];
+  OCMVerify([self.mockSharedInstance setClientID:@"mockClientId"]);
+}
+
+#pragma mark - Is signed in
+
+- (void)testIsNotSignedIn {
+  OCMStub([self.mockSharedInstance hasPreviousSignIn]).andReturn(NO);
+
+  FlutterMethodCall *methodCall = [FlutterMethodCall methodCallWithMethodName:@"isSignedIn"
+                                                                    arguments:nil];
+
+  XCTestExpectation *expectation = [self expectationWithDescription:@"expect result returns true"];
+  [self.plugin handleMethodCall:methodCall
+                         result:^(NSNumber *result) {
+                           XCTAssertFalse(result.boolValue);
+                           [expectation fulfill];
+                         }];
+  [self waitForExpectationsWithTimeout:5.0 handler:nil];
+}
+
+- (void)testIsSignedIn {
+  OCMStub([self.mockSharedInstance hasPreviousSignIn]).andReturn(YES);
+
+  FlutterMethodCall *methodCall = [FlutterMethodCall methodCallWithMethodName:@"isSignedIn"
+                                                                    arguments:nil];
+
+  XCTestExpectation *expectation = [self expectationWithDescription:@"expect result returns true"];
+  [self.plugin handleMethodCall:methodCall
+                         result:^(NSNumber *result) {
+                           XCTAssertTrue(result.boolValue);
+                           [expectation fulfill];
+                         }];
+  [self waitForExpectationsWithTimeout:5.0 handler:nil];
+}
+
+#pragma mark - Sign in silently
+
+- (void)testSignInSilently {
+  OCMExpect([self.mockSharedInstance restorePreviousSignIn]);
+
+  FlutterMethodCall *methodCall = [FlutterMethodCall methodCallWithMethodName:@"signInSilently"
+                                                                    arguments:nil];
+
+  [self.plugin handleMethodCall:methodCall
+                         result:^(id result){
+                         }];
```

Comment:
Same question here. (And all following cases that don't assert anything about the return value)

Reply:
Same comment as above, in GoogleSignIn 6.0 this became a callback `restorePreviousSignInWithCallback:` instead of a delegate.
https://github.com/flutter/plugins/blob/b548ae18646d1725a8e1f004c26bf42e4d362aa8/packages/google_sign_in/google_sign_in/ios/Classes/FLTGoogleSignInPlugin.m#L103-L106
is handled in the delegate method:
https://github.com/flutter/plugins/blob/b548ae18646d1725a8e1f004c26bf42e4d362aa8/packages/google_sign_in/google_sign_in/ios/Classes/FLTGoogleSignInPlugin.m#L208-L214
will become the easily testable:
```obj-c
    [GIDSignIn.sharedInstance restorePreviousSignInWithCallback:^(GIDGoogleUser *user, NSError *error) {
      [self didSignInForUser:user result:result withError:error];
    }];
```